buildSrc/src/gen-execution.kt

```diff
@@ -5,6 +5,205 @@ import org.intellij.lang.annotations.Language
+			super(type);
+		}
+		
+		@NotNull
+		@Override
+		public ${languageName}RunConfiguration createTemplateConfiguration(@NotNull Project project) {
+			return new ${languageName}RunConfiguration(project, this);
+		}
+	}
+	""".trimIndent()
+
+	@Language("java")
+	val runConfigTypeJava = """
+	package $basePackage.execution;
+
+	import com.intellij.execution.configurations.ConfigurationFactory;
+	import com.intellij.execution.configurations.ConfigurationType;
+	import icons.TTIcons;
+	import ${basePackage}.TTBundle;
+	import org.jetbrains.annotations.Nls;
+	import org.jetbrains.annotations.NotNull;
+
+	import javax.swing.*;
+
+	public class ${languageName}RunConfigurationType implements ConfigurationType {
+
+		ConfigurationFactory factory;
+		private ConfigurationFactory[] factories;
+		private static ${languageName}RunConfigurationType instance;
+		
+		synchronized public static ${languageName}RunConfigurationType getInstance() {
+			if (instance == null) {
+				instance = new ${languageName}RunConfigurationType();
+			}
+			return instance;
+		}
+
+		private ${languageName}RunConfigurationType() {
+			factory = new ${languageName}RunConfigurationFactory(this);
+			factories = new ConfigurationFactory[]{factory};
+		}
+
+		@NotNull
+		@Override
+		public String getDisplayName() {
+			return TTBundle.message("$nickname.name");
+		}
+
+		@Nls
+		@Override
+		public String getConfigurationTypeDescription() {
+			return TTBundle.message("$nickname.run-config.description");
+		}
+
+		@Override
+		public Icon getIcon() {
+			return TTIcons.$constantPrefix;
+		}
+
+		@NotNull
+		@Override
+		public String getId() {
+			return "${constantPrefix}_RUN_CONFIG_ID";
+		}
+
+		@Override
+		public ConfigurationFactory[] getConfigurationFactories() {
+			return factories;
+		}
+	}
+		
+	""".trimIndent()
+
+	@Language("java")
+	val runConfigJava = """
+	package $basePackage.execution;
+	
+	import com.intellij.execution.ExecutionException;
+	import com.intellij.execution.Executor;
+	import com.intellij.execution.configurations.ConfigurationFactory;
+	import com.intellij.execution.configurations.RunConfiguration;
+	import com.intellij.execution.configurations.RunProfileState;
+	import com.intellij.execution.runners.ExecutionEnvironment;
+	import com.intellij.openapi.options.SettingsEditor;
+	import com.intellij.openapi.project.Project;
+	import com.intellij.openapi.util.JDOMExternalizerUtil;
+	import ${basePackage}.TTBundle;
+	import ${basePackage}.execution.${languageName}CommandLineState;
+	import ${basePackage}.execution.${languageName}RunConfigurationEditor;
+	import ${basePackage}.execution.InterpretedRunConfiguration;
+	import org.jdom.Element;
+	import org.jetbrains.annotations.NotNull;
+	import org.jetbrains.annotations.Nullable;
+
+	public class ${languageName}RunConfiguration extends InterpretedRunConfiguration<${languageName}CommandLineState> {
+		public String ${configName}Executable;
+		private Project project;
```

Comment:
I believe the superclass has got a `getProject` method. Btw, why is the `factory` needed?

Reply:
Will delete project and factor field.